R/reference.R

```diff
@@ -18,9 +18,17 @@
   files <- list.files("man", full.names = TRUE)
```

Comment:
When I render `marginaleffects`, the two top functions are: `comparisons` (which is exported; `@export`), and `complete_levels` (which is not exported; `@keywords internal`). I see. Looking at `Rd` files in `man`, I don't see an export tag in `comparisons.Rd`. The only tag that seems preserved is `\keyword{internal}`. Do you want me to keep exported files using a negative logic (exclude those with `\keyword{internal}`), knowing that not everybody uses this? Or do you have a better idea? One possibility is to parse the R files looking for the export tag, but should there not be an easier solution?

Reply:
When I render `marginaleffects`, the two top functions are: `comparisons` (which is exported; `@export`), and `complete_levels` (which is not exported; `@keywords internal`). I see. Looking at `Rd` files in `man`, I don't see an export tag in `comparisons.Rd`. The only tag that seems preserved is `\keyword{internal}`. Do you want me to keep exported files using a negative logic (exclude those with `\keyword{internal}`), knowing that not everybody uses this? Or do you have a better idea? One possibility is to parse the R files looking for the export tag, but should there not be an easier solution?

Reply:
No we don't want our solution to be roxygen2 specific. We need this to work with other packages too, and parsing is error prone.

There are many options like looking at NAMESPACE or this function: https://stackoverflow.com/a/9675792/342331

Reply:
Ah, great, thanks!

Reply:
Ok, I got something running, but the problem is that for example, for `altdoc`, the file `init.Rd` contains all three `use_*` functions. So even though I can get the list of exported functions through `getNamespaceExports(pkg)`, they don't match the `.Rd` file names. So it seems like I would still need to somehow select all `.Rd` files, then sub-select by parsing each `.Rd` file to see if they include the exported function names from `getNamespaceExports(pkg)`...

Reply:
Sure, just intersect() them.

Reply:
I don't think `intersect()` would work because that is like an inner joint: it only keeps elements present in both lists. But in the case of the `init.Rd` file, it doesn't match with any of the exported function names. Right now I'm trying some regex within the `.Rd` with `\name{`...

Reply:
Ok, not pretty but the following seems to work...

```r
files <- list.files("man", pattern = ".Rd", full.names = TRUE)
pkg <- basename(getwd())
exported <- getNamespaceExports(pkg)

which.files <- lapply(files, function(x) {
    y <- readLines(x)
    y <- grep("\\name{", y, fixed = TRUE, value = TRUE)
    y <- gsub("\\name{", "", y, fixed = TRUE)
    y <- gsub("}", "", y, fixed = TRUE)
    y %in% exported
})

files <- files[unlist(which.files)]
```

![only exported](https://github.com/etiennebacher/altdoc/assets/13123390/2e142207-658d-4d60-ac49-92517d0448e1)

Reply:
Hmm, that's weird. I didn't expect all these `get_()` and `expect_()` functions to be exported. They are not user-facing. I guess as long as there's a way to manually set them, it's fine like that. Let me know when you'd like me to review.

Reply:
But in the file `get_coef.R`, you have

```r
#' @export
get_coef <- function (model, ...)
```

And so it is exported:

``` r
getNamespaceExports("marginaleffects")[32]
#> [1] "get_coef"
```

<sup>Created on 2023-11-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

After the last push I got an error in tests, so I'll try fixing that and then it should be ready for review. I will let you know.

Reply:
Yeah, you're right. I forgot about that. I'll still be able to specify manually, right?

Reply:
`use_docute()` and family have a `custom_reference` argument: 
```
#' @param custom_reference Path to the file that will be sourced to generate the
#' "Reference" section.
```
So yes :) I think if you provide your custom Reference file, it should work!